packages/nest/README.md

```diff
@@ -70,27 +99,13 @@ import { BooleanFeatureFlag } from '@openfeature/nestjs-sdk';
 import { EvaluationDetails } from '@openfeature/server-sdk';
 import { Request } from 'express';
 
-function getContext(executionContext: ExecutionContext) {
-  const request = executionContext.switchToHttp().getRequest<Request>();
-  const userId = request.header('x-user-id');
-
-  if (!userId) {
-    return undefined;
-  }
-
-  return {
-    targetingKey: userId,
-  };
-}
-
 @Controller()
 export class OpenFeatureController {
   @Get('/welcome')
   public async welcome(
     @BooleanFeatureFlag({
       flagKey: 'testBooleanFlag',
       defaultValue: false,
-      contextFactory: getContext,
     })
       feature: Observable<EvaluationDetails<boolean>>,
```

Comment:
Can this inject an observable instead of a promise? Personally, I don't love working with observables, especially if you're not dealing with a data stream.

FYI @lukas-reining

Reply:
You mean a `Promise` instead of an `Observable` @beeme1mr?
I am fine with using a promise, but my feeling is that Nest.js tends more to use observables for such cases. (e.g. https://github.com/open-feature/js-sdk-contrib/tree/main/libs/providers/config-cat#installation)

Reply:
haha, yeah.

Reply:
Thinking about this a bit more, I'm not sure if there's a use case for injecting the value of an individual feature flag. It seems much more likely that a user would inject the client into the class.

Reply:
@beeme1mr Ya that link was wrong, changed it😅

Regarding the feature:
I could imagine that e.g. changing if the endpoint should be "enabled" or changing the whole request would be interesting.

Reply:
I've already done something pretty similar to this at controller level changing from older to newer/refactored version. Even though I'm not a big fan of observables either, it might be useful.

Reply:
@beeme1mr What do you think? I still feel Observables are the most idiomatic for Nest but if others are leaning more to Promises, I would be fine changing it.

Reply:
I'm fine with using an Observable here. 👍